### PR TITLE
AKS Diagnostics Enable CRUD detectors

### PR DESCRIPTION
**Pull request by Tatsinnit**
This is PR is open for enabling the `CRUD Detector` via `Vscode` -> `Run AKS Diagnostics`.

This work enable 2 things:

* Submenu - feature for `Run AKS Diagnostics`. 
* On `one-click` the `ARM call gets made to AppLens API` and we populate the data for CRUD related things for the cluster, along with `Clickable portal link`.

Thanks heaps,

cc: @rzhang628, @squillace, @itowlson, @gambtho , @davefellows as fyi next vscode feature.

image

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ Right click on your AKS cluster and click on **Save Kubeconfig** to save the kub
...
+Right click on your AKS cluster and click on **Run AKS Diagnostics** which will expand submenu options clusters network and connectivity issues or Create, Upgrade, Delete and Scale issues of your cluster to run a series of checks on your AKS cluster's backend telemetry to get diagnostics information. Also, to perform more checks on your AKS cluster to troubleshoot and get recommended solutions, click on the AKS Diagnostics link at the top of the page to open it for the selected cluster. For more information on AKS Diagnostics, visit [AKS Diagnostics Overview](https://docs.microsoft.com/azure/aks/concepts-diagnostics).
```
Comment (peterbom) [inline README.md:45]: Possible alternative:
```
Right click on your AKS cluster and click on **Run AKS Diagnostics** to display diagnostics information based on your AKS cluster's backend telemetry for:
- Create, Upgrade, Delete and Scale issues
- Network and Connectivity issues

To perform further checks on your AKS cluster to troubleshoot and get recommended solutions, click on the AKS Diagnostics link at the top of the page to open it for the selected cluster. For more information on AKS Diagnostics, visit [AKS Diagnostics Overview](https://docs.microsoft.com/azure/aks/concepts-diagnostics). 
```

Comment (Tatsinnit) [reply README.md:45]: ☕️👍 💯 % , sorry I wrote what came to my mind and sounded right but this is much better. Sounds awesome, I’ll add it soon.

File: src/commands/aksCRUDDiagnostics/aksCRUDDiagnostics.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+async function loadDetector(
+    cloudTarget: AksClusterTreeItem,
+    extensionPath: string) {
+    const clustername = cloudTarget.name;
+
+    await longRunning(`Loading ${clustername} diagnostics.`,
+          async () => {
+            const clusterAppLensData = await getAppLensDetectorData(cloudTarget, "aks-category-crud");
+            const detectorMap = new Map();
```
Comment (peterbom) [inline src/commands/aksCRUDDiagnostics/aksCRUDDiagnostics.ts:42]: See suggestions in https://github.com/Tatsinnit/vscode-aks-tools/pull/3/files

Not major, and could be a separate PR, but given this is a new file it'd maybe make sense to keep it as part of this one, if they make sense!

**Review by peterbom — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Tatsinnit) [reply README.md]: 💡☕️ changes done. Thank you and also adding @rzhang628 for quick read and approval and then I will go ahead and merge this one.

**Comment by rzhang628** [comment]
Looks great to me! Approved.

**Review by rzhang628 — APPROVED** [review]
Looks great to me, approved!

File: src/commands/aksCRUDDiagnostics/aksCRUDDiagnostics.ts (hunk 2 above)
Comment (Tatsinnit) [reply src/commands/aksCRUDDiagnostics/aksCRUDDiagnostics.ts:42]: 💡☕️ cool, let’s merge this any other refactor we can do that as separate refactor so that feature PR stays more focused towards aks diagnostic crud feature. 😊
